lib/AST/TypeJoinMeet.cpp

```diff
@@ -301,14 +304,23 @@ CanType TypeJoin::visitDependentMemberType(CanType second) {
 CanType TypeJoin::visitFunctionType(CanType second) {
   assert(First != second);
 
-  if (First->getKind() != second->getKind())
-    return TheAnyType;
+  auto secondFnTy = second->castTo<FunctionType>();
+
+  if (First->getKind() != second->getKind()) {
+    if (secondFnTy->getExtInfo().isNoEscape()) {
+      return Nonexistent;
```

Comment:
Why not just return CanType()?

Reply:
As the comment for this member calls out, this makes it clear why the value is being returned.